bin/openshift-include.sh

```diff
@@ -4,6 +4,73 @@
 # This script is not intended to be run directly
 # Assumes bin/common.sh has been sourced
 
+function ocVersionCheck {
+  origIFS=$IFS
+  IFS=$'\n'
+
+  allArr=($(oc version 2>/dev/null))
+  IFS=$origIFS
+
+  for (( i=0; i<${#allArr[@]}; i++ )); do
+    # Split the line into an array
+    verArr=(${allArr[$i]})
+    if [ ${#verArr[@]} -eq 3 ]; then
+      verType="${verArr[0]}"
+      ver="${verArr[2]}"
+      if [ "$verType" == "Client" ]; then
+          ver="${verArr[2]}"
+          if [[ $ver =~ v(([0-9]+)\.([0-9]+)\.([0-9]+)) ]]; then
+            OC_FULL_VERSION=${BASH_REMATCH[1]}
+            OC_MAJOR_VERSION=${BASH_REMATCH[2]}
+            OC_MINOR_VERSION=${BASH_REMATCH[3]}
+            OC_PATCH_VERSION=${BASH_REMATCH[4]}
+          else
+            echo "Unable to parse client version: [$ver]"
+          fi
+      elif [ "$verType" == "Server" ]; then
+          ver="${verArr[2]}"
+          if [[ $ver =~ (([0-9]+)\.([0-9]+)\.([0-9]+)) ]]; then
+            OSHIFT_FULL_VERSION=${BASH_REMATCH[1]}
+            OSHIFT_MAJOR_VERSION=${BASH_REMATCH[2]}
+            OSHIFT_MINOR_VERSION=${BASH_REMATCH[3]}
+            OSHIFT_PATCH_VERSION=${BASH_REMATCH[4]}
+          else
+            echo "Unable to parse server version: [$ver]"
+          fi
+      fi
+    fi
+  done
+  log_info "OpenShift client version: $OC_FULL_VERSION"
+  log_info "OpenShift server version: $OSHIFT_FULL_VERSION"
+  # Version enforcement
+  if [ "$OPENSHIFT_VERSION_CHECK" == "true" ]; then
+    if [ "$OC_MAJOR_VERSION" -lt 4 ]; then
+      log_error "Unsupported 'oc' version: $OC_FULL_VERSION. Version 4+ is required."
+      exit 1
+    fi
+    if [ "$OSHIFT_MAJOR_VERSION" -eq 4 ] && [ "$OSHIFT_MINOR_VERSION" -eq 6 ]; then
+      log_debug "OpenShift version check OK"
+    elif [ "$OSHIFT_MAJOR_VERSION" -lt 4 ]; then
+      log_error "Unsupported OpenShift version: $OSHIFT_FULL_VERSION"
+      log_error "Supported versions include: 4.6.x"
```

Comment:
Why the vague wording?  You're throwing an error if it isn't at least 4.6...seems like the supported version is 4.6+

Reply:
See latest changes